doc/sphinx/source/vulkan.rst

```diff
@@ -0,0 +1,126 @@
+Vulkan driver
+=================
+
+NOTE: THIS DRIVER IS INCOMPLETE AND IS NOT ACTIVELY DEVELOPED BY TUNI.
+ANY MAINTENANCE WOULD NEED TO BE DONE BY COMMUNITY IN THEIR FREE TIME.
+
+PULL REQUESTS ON GITHUB WILL BE CONSIDERED, BUT REPORTS ON
+MISSING FEATURES WITHOUT CODE CONTRIBUTIONS ARE POINTLESS,
+BECAUSE LIMITATIONS ARE ALREADY KNOWN.
+
+This driver uses libvulkan and clspv to run OpenCL code on GPU devices via Vulkan API.
+
+Installation
+-------------
+
+Required:
+
+ * vulkan drivers (on Ubuntu, "mesa-vulkan-drivers" for opensource vulkan drivers)
+ * vulkan development files (on Ubuntu, "vulkan-headers" and "libvulkan-dev")
+ * SPIR-V tools (for clspv; on Ubuntu, package "spirv-tools")
+
+Optional:
+
+ * "vulkan-validationlayers-dev" for vulkan validation layers
+ * "vulkan-tools" or "vulkan-utils" package for vulkaninfo
+
+Note that the Vulkan device MUST support the following extensions (clspv requirements):
+
+* VK_KHR_variable_pointers
+* VK_KHR_storage_buffer_storage_class
+* VK_KHR_shader_non_semantic_info
+
+Easiest to check is with vulkaninfo utility, they must be listed in 'Device Extensions' section.
+
+To build the full pocl-vulkan, first you must build the clspv compiler::
+
+    git clone https://github.com/google/clspv.git
+    cmake /path/to/clspv -DCLSPV_BUILD_TESTS=OFF -DCMAKE_INSTALL_PREFIX=${CLSPV_BIN_DIR} -DCMAKE_BUILD_TYPE=Release
+    make -jX
+    make install
+
+... this will take some time and space, because it compiles its own checkout of LLVM.
+
+After the build, copy "clspv" and "clspv-reflection" to some place CLSPV_BIN_DIR
+
+Then build the vulkan driver::
+
+    cmake -DENABLE_HOST_CPU_DEVICES=0 -DENABLE_LLVM=0 -DENABLE_EXPERIMENTAL_DRIVERS=1 -DENABLE_VULKAN=1 -DCLSPV=${CLSPV_BIN_DIR}/clspv <path-to-pocl-source-dir>
```

Comment:
I accidentally omitted /clspv and it crashed at runtime due to a stack smashing detector :) Not sure what to do to avoid it as once might want to build without the clspv being there at build time. Warn if it's not there?

Reply:
Did you do something for this?

Reply:
CMake now checks the existence of clspv binary at cmake time. You can also compile the vulkan driver without clspv. I did not add any runtime checking of clspv existence, should i add it ?

Reply:
Not needed. This is a very experimental and incomplete driver anyhow.